src/prototype_creep_routing.js

```diff
@@ -309,8 +309,21 @@ Creep.prototype.moveByPathMy = function(route, routePos, start, target, skipPreM
     // this.pos.getDirectionTo(search.path[0]) + ' pathPos: ' + pathPos + ' pos:
     // ' + this.pos + ' routePos: ' + routePos + ' path: ' +
     // JSON.stringify(path) + ' route: ' + JSON.stringify(route));
-    this.say('R:p-1: ' + this.pos.getDirectionTo(search.path[0]), true);
-    let returnCode = this.move(this.pos.getDirectionTo(search.path[0]));
+    this.say('R:p-1: ');
+    //let closestPathRange = 50;
+    //let closestPathPosition;
+    //for (let iterator = 0; iterator < search.path.length; iterator++) {
+    //  if (this.pos.getRangeTo(search.path[iterator] < closestPathRange)) {
+    //    closestPathPosition = iterator;
+    //    closestPathRange = this.pos.getRangeTo(search.path[iterator] < closestPathRange);
+    //  }
+    //}
+
+    let returnCode = this.moveTo(_.min(search.path, function(object) {
```

Comment:
It doesn't matter, but I tried to pass always the position to method like this. Also with the background to overwrite the method and only have to handle positions.

Much better.